docs/testing/workspaces-migration-guide.md

```diff
@@ -295,10 +289,13 @@ assert_eq!(outcome.logs[0], "The account of the sender was deleted");
 assert_eq!(outcome.logs[2], format!("Account @{} burned {}", contract.id(), 10));
 ```
 
-## Profiling gas
+## Profiling Gas
 
-NEAR Simulator has not been providing realistic gas estimates (TODO: needs elaboration - since when and why?), so the `workspaces-rs` gas estimates might come as a surprise. The good news is that they do reflect reality!
+NEAR Simulator never had accurate gas estimations since it only tried to mirror nearcore, but nearcore has extra functionality on top which consumes gas (like cross-contract calls are processed separately from the same transaction and that incurs gas fees). Workspaces offers the better experience here and aligns very well with what you can do on testnet and mainnet. It provides the added benefit of allowing the developer to accurately profile gas usage before deploying to `mainnet`.
 
+:::warning
+Since `workspaces-rs` is now using accurate gas measurements, some contracts that were previously being tested with sdk-sim might not work the same. You should do your due diligence if you plan to deploy to `mainnet`.
```

Comment:
I think the wording here is a bit inaccurate. The contracts will work the same, it is just the existing flows might break if the user was depending on sdk-sim gas reports. For example, imagine if a user had a script that was doing automatic deployments only if (tested) gas usage was less than X TGas and it was always so, but now with realistic gas estimates it is more than X so the scripts will stop deploying the contract.